lib/skate_web/auth_manager.ex

```diff
@@ -1,28 +1,56 @@
 defmodule SkateWeb.AuthManager do
   use Guardian, otp_app: :skate
+  alias Skate.Settings.Db.User, as: DbUser
+  alias Skate.Settings.User
+  require Logger
 
   @type access_level :: :none | :general | :admin
 
   @skate_admin_group "skate-admin"
   @skate_dispatcher_group "skate-dispatcher"
+  @v2_resource_prefix "v2:"
+
+  def v2_resource_prefix, do: @v2_resource_prefix
+
+  def subject_for_token(%DbUser{id: user_id}, _claims) do
+    {:ok, "#{@v2_resource_prefix}#{user_id}"}
+  end
 
   def subject_for_token(resource, _claims) do
+    Logger.info("old pattern matched: subject_for_token")
     {:ok, resource}
   end
 
+  def resource_from_claims(%{"sub" => @v2_resource_prefix <> user_id}) do
+    case User.get_by_id(String.to_integer(user_id)) do
```

Comment:
I'm having second thoughts about introducing a DB query here. It seems to be a conventional approach based on the guardian docs linked in the PR description, but based on initial logging, it seems that this query will be performed on every endpoint hit. 

I'm looking into a way to avoid that query (resource should be cached on the token - why do we need to re-call this?). If not, we could avoid the query by passing username as part of "sub", but that seems a bit like a smell to me.

Reply:
After some research into 2 (doesn't seem viable) and discussion with Eddie, going to pursue option 1. 

Removing the query will help minimize the impact of this change to keep it focused on only changing token format rather than changing token & introducing new frequent DB request.